src/components/CodeEditor.tsx

```diff
@@ -65,6 +75,26 @@ export const CodeEditor = ({
     }
   }, [initialCode])
 
+  useEffect(() => {
+    if (manualEditsJson) {
+      setFiles((prev) => ({
+        ...prev,
+        "manual-edits.json": manualEditsJson,
+      }))
```

Comment:
technically this is a re-render anti-pattern, if `files` is completely derived you should use `useMemo` for it rather than the antipattern "`useEffect` with `setState`" which causes unnecessary re-renders and can complicate the rendering flow.

Can you say something like:

```tsx
const files = useMemo(() => ({
  "index.tsx": code,
  "manual-edits.json": manualEditsJson
}, [code, manualEditsJson])
```

`useMemo` can't cause re-renders, so it's much safer.